### PR TITLE
Dockerize the Big Data Crawler Server

### PR DESCRIPTION
**Pull request by ramesh-maddegoda**
## 🗒️ Summary
The pull request to dockerize the Big Data Crawler Server 

## ♻️ Related Issues
Refer to the task NASA-PDS/pds-registry-app#200

### CODE REVIEW

File: docker/README.md [hunk 1]
```diff
@@ -0,0 +1,63 @@
...
+# Set the following argument with a compatible Big Data Crawler Server version
+ARG big_data_crawler_server_version=1.0.0-SNAPSHOT
+```
+
+#### 2. Open a terminal and change the current working directory to `big-data-crawler-server/docker`.
+
+#### 3. Build the docker image as follows.
+
+```
+docker image build -t nasapds/big-data-crawler-server .
```
Comment (nutjob4life) [inline docker/README.md]: In documentation (like READMEs), I like to see full arguments always: `-t` → `--tag` … but that's just my personal preference 😇

File: docker/Dockerfile [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
+# POSSIBILITY OF SUCH DAMAGE.
+
+FROM openjdk:11-slim
+
+# Set the following argument with a compatible Big Data Crawler Server version
+ARG big_data_crawler_server_version=1.0.0-SNAPSHOT
+
+# Install curl
+RUN apt-get update -y
+RUN apt-get install curl -y
```
Comment (nutjob4life) [inline docker/Dockerfile]: I think it's a good practice to minimize the number of `RUN` statements in a Dockerfile as this reduces image size. Cleaning up after installing packages also helps. You might consider coalescing these multiple `RUN` statements into a single `RUN` with something like:
```Dockerfile
RUN  apt-get update -y &&\
     apt-get install curl -y &&\
     mkdir /opt/big-data-crawler-server &&\
     tar xzf /tmp/big-data-crawler-server-bin.tar.gz  -C /opt/big-data-crawler-server --strip-components 1 &&\
     rm -f /tmp/big-data-crawler-server-bin.tar.gz &&\
     apt-get autoclean --quiet --yes &&\
     rm --force --recursive /var/lib/apt/lists/*
```

File: docker/run.sh [hunk 3]
```diff
@@ -0,0 +1,70 @@
...
+            "as the environment variable 'HARVEST_DATA_DIR'." 1>&2
+    exit 1
+fi
+
+# Execute docker container run to start the Big Data Crawler Server
+docker container run --name big-data-crawler-server \
+           --rm \
+           --volume "$CRAWLER_SERVER_CONFIG_FILE":/cfg/crawler-server.cfg \
+           --volume "$HARVEST_DATA_DIR":/data \
+           nasapds/big-data-crawler-server
```
Comment (nutjob4life) [inline docker/run.sh:70]: Looks fantastic

**Review by nutjob4life — APPROVED** [review]
One minor image size comment otherwise looks good

**Review by tloubrieu-jpl — APPROVED** [review]
@ramesh-maddegoda you can update as recommanded by @nutjob4life to minimize the number of call to RUN if that makes sense to you.

File: docker/README.md (hunk 1 above)
Comment (ramesh-maddegoda) [reply docker/README.md]: Agreed! Updated the READMEs in all 3 related pull requests.

File: docker/Dockerfile (hunk 2 above)
Comment (ramesh-maddegoda) [reply docker/Dockerfile]: Thanks! Later, I realized that now it is not necessary to have curl installed. However, I followed the coding style in your example with the remaining code.

**Review by tloubrieu-jpl — APPROVED** [review]

**Pull request merged**